src/resolver.ts

```diff
@@ -188,8 +188,13 @@ export class Resolver extends DiagnosticEmitter {
     // Look up in contextual types if a simple type
     if (isSimpleType) {
       let simpleName = nameNode.identifier.text;
+      let type: Type | null = null;
       if (ctxTypes && ctxTypes.has(simpleName)) {
-        let type = assert(ctxTypes.get(simpleName));
+        type = assert(ctxTypes.get(simpleName));
+      } else {
+        type = ctxElement.lookupGenericType(simpleName);
+      }
+      if (type) {
```

Comment:
This looks like there is a foregoing issue, as `ctxTypes` should have all the types. Is there perhaps something missing instead where `ctxTypes` is populated? Like, right now, `lookupGenericType` does after-the-fact what should already have been done before it seems.

Reply:
Yes, we just need to find a position to sync generic type in `ctxElement` to `ctxType` when `ctxElement` is `Function` or `Class`.